lib/lineup/src/lib.rs

```diff
@@ -26,5 +26,21 @@ pub mod upcalls;
 /// Type to represent a core id for the scheduler.
 type CoreId = usize;
 
+/// A utility function that converts a core_id (which may not be sequential)
+/// to a form more easily used for indexing
+#[inline(always)]
+pub fn core_id_to_index(core_id: CoreId) -> usize {
+    let machine_id = kpi::system::mid_from_gtid(core_id);
+    if machine_id == 0 {
+        // This is controller (should never happen) or non-rackscale
```

Comment:
maybe panic!()?

Reply:
There's no build flag in the userspace programs (I don't think) to determine if it's rackscale or not. So I don't think there's a good way to determine between the two cases, but something very wrong would have to happen in quite a few other places in order to be running userspace code on the controller.